src/Rules/Generate.hs

```diff
@@ -177,6 +185,31 @@ generateRules = do
   where
     file <~ gen = file %> \out -> generate out emptyTarget gen
 
+-- | Generate scripts the build system requires. For now we generate the
+-- @ghc-split@ script from it's literate perl source.
+generateScripts :: Rules ()
+generateScripts = do
+    -- how to translate literate perl to perl.
+    -- this is a hack :-/
+    "//*.prl" %> \out -> do
```

Comment:
I think the rules responsible for working with `perl` should live in a separate file `Rules/Perl.hs`. Semantically they are not generators, they are invocations of a builder `Unlit` on a source file to produce an output file.